stablehlo/tests/stablehlo_legalize_quant_to_int.mlir

```diff
@@ -0,0 +1,2229 @@
+// RUN:  stablehlo-opt --stablehlo-legalize-quant-to-int -split-input-file %s -verify-diagnostics | FileCheck %s
+
+// CHECK-LABEL: func @uniform_quantize_and_dequantize
```

Comment:
Can we add one hand-written test to reference that uses this pass? Will help for discovery of someone with the use case "how do I interpret programs with uq types?" Could be `uniform_quantize.mlir` and have a basic UQ add DQ test

Reply:
The pipeline is embedded in interpreter pipeline https://github.com/openxla/stablehlo/pull/2388 and added the necessary tests.